packages/core/src/manifest.ts

```diff
@@ -180,7 +262,10 @@ export class Manifest {
   }
 
   private async lock() {
-    const lockfileName = path.join(manifestDir, `chain-${this.chainId}`);
+    const lockfileName = path.join(
+      this.devDir ?? MANIFEST_DEFAULT_DIR,
```

Comment:
Might be better to keep a non-optional `dir` variable in both cases so we use it consistently here and in the constructor.

```suggestion
      this.dir,
```